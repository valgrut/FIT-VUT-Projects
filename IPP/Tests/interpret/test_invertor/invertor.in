Tohle\032je\032muj\032test\032s\032\092\032backslashem.
