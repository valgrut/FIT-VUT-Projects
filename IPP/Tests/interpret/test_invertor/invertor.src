.IPPcode18

DEFVAR GF@input
DEFVAR GF@result
DEFVAR GF@len
DEFVAR GF@tmp

MOVE GF@result string@

WRITE string@Zadej\032Retezec\032k\032inverznuti
READ GF@input string

STRLEN GF@len GF@input
JUMPIFEQ end GF@len int@0
SUB GF@len GF@len int@1

LABEL inverse
GETCHAR GF@tmp GF@input GF@len
CONCAT GF@result GF@result GF@tmp
JUMPIFEQ end GF@len int@0
SUB GF@len GF@len int@1
JUMP inverse

LABEL end
write string@VÃ½sledek:\032
WRITE GF@result
write string@   #novy radek

WRITE string@zkusim\032malou\032upravu:\032
SETCHAR GF@result int@0 string@A 
WRITE GF@result
