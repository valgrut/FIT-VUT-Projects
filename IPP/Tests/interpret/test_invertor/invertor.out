0
Zadej Retezec k inverznuti
VÃ½sledek:
.mehsalskcab \ s tset jum ej elhoT

zkusim malou upravu:
Amehsalskcab \ s tset jum ej elhoT
